sympy/solvers/solveset.py

```diff
@@ -1192,3 +1196,837 @@ def linsolve(system, *symbols):
+    if intersections and complements:
+        # no testcase is added for this block
+        result_all_variables = add_intersection_complement(
+            result_all_variables, intersections,
+            Intersection=True, Complement=True)
+    elif intersections:
+        result_all_variables = add_intersection_complement(
+            result_all_variables, intersections, Intersection=True)
+    elif complements:
+        result_all_variables = add_intersection_complement(
+            result_all_variables, complements, Complement=True)
+
+    # convert to ordered tuple
+    result = S.EmptySet
+    for r in result_all_variables:
+        temp = [r[symb] for symb in all_symbols]
+        result += FiniteSet(tuple(temp))
+    return result
+# end of def substitution()
+
+
+def _solveset_work(system, symbols):
+        soln = solveset(system[0], symbols[0])
+        if isinstance(soln, FiniteSet):
+            _soln = FiniteSet(*[tuple((s,)) for s in soln])
+            return _soln
+        else:
+            return FiniteSet(tuple(FiniteSet(soln)))
+
+
+def _handle_positive_dimensional(polys, symbols, denominators):
+    from sympy.polys.polytools import groebner
+    # substitution method where new system is groebner basis of the system
+    _symbols = list(symbols)
+    _symbols.sort(key=default_sort_key)
+    basis = groebner(polys, _symbols, polys=True)
+    new_system = []
+    for poly_eq in basis:
+        new_system.append(poly_eq.as_expr())
+    result = [{}]
+    result = substitution(
+        new_system, symbols, result, [],
+        denominators)
+    return result
+# end of def _handle_positive_dimensional()
+
+
+def _handle_zero_dimensional(polys, symbols, system):
+    # solve 0 dimensional system using `solve_poly_system`
+    result = solve_poly_system(polys, *symbols)
+    # May be some extra soln is added because
+    # we did `poly = g.as_poly(*symbols, extension=True)`
+    # need to check that and remove if not a soln
+    result_update = S.EmptySet
+    for res in result:
+        dict_sym_value = dict(list(zip(symbols, res)))
+        if all(checksol(eq, dict_sym_value) for eq in system):
+            result_update += FiniteSet(res)
+    return result_update
+# end of def _handle_zero_dimensional()
+
+
+def _separate_poly_nonpoly(system, symbols):
+    polys = []
+    polys_expr = []
+    nonpolys = []
+    denominators = set()
+    for eq in system:
+        # Store denom expression if it contains symbol
+        denominators.update(_simple_dens(eq, symbols))
+        # try to remove sqrt and rational power
+        without_radicals = unrad(simplify(eq))
+        if without_radicals:
+            eq_unrad, cov = without_radicals
+            if not cov:
+                eq = eq_unrad
+        if isinstance(eq, Expr):
+            eq = eq.as_numer_denom()[0]
+            # this will remove sqrt if symbols are under it
+            poly = eq.as_poly(*symbols, extension=True)
+        if poly is not None:
```

Comment:
I think it should check for `Number` also, if `eq` is a Number then eq will not be added in `poly` and `nonpoly` list ( since that will not help for finding soln). It seems I forgot to add line `poly = None` at the starting. I have edited this in new commit. Thanks.